### ISSUE TITLE
Probability calibration

### ISSUE THREAD

**Issue by tabacof** [Accenture/AmpliGraph#131]
**Background and Context**
A prediction is calibrated if the probabilities make sense in frequentist terms. A prediction of 95% should be true 95% of the times. AmpliGraph models output a real-valued score that could be turned into a probability by a sigmoid transform, but the resulting probability wouldn't be calibrated.

Calibration is highly dependent on the ratio of positive to negatives, that is the main challenge in knowledge graphs, where in many times you don't have explicit negatives. In those cases, negatives can be generated by subject or object corruptions. The choice of entities to consider in the negative generation will have great impact in the calibration probabilities.

**Description**
Adding a new method to the baseline embedding model that receives true and negative triples as inputs and calibrates the score using Platt-scaling (more easily implemented in TensorFlow than isotonic regression). If the user does not provide the negatives, automatic generation options will be provided (similar to the evaluation protocol).

**Comment by tabacof** [comment]
Results for Platt scaling calibration on the Wordnet11 dataset (where we have positive and negative triples), calibrated on a small validation set:
![image](https://user-images.githubusercontent.com/2151702/63601319-1ac32c00-c5bd-11e9-8d13-2d753867dfd4.png)

We can see that:
1. Applying the expit (sigmoid) to the scores as we do in the basic tutorial leads to terrible calibration results, with the exception being the NLL loss (which is a proper scoring rule)
2. Calibration consistently deliver results better than the baseline
3. The variance of the results after calibration is much smaller than the variance of the results before, but a better original result leads to better calibration all the time

**Comment by tabacof** [comment]
Solved by #136
